### PR TITLE
Factor out getRosDistro

### PR DESCRIPTION
**Pull request by mvukov**
### Public-Facing Changes

None.

### Description

Robustified fetching of ROS_DISTRO env var.

### CODE REVIEW

**Comment by mvukov** [comment]
@achim-k PTAL.

File: ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp [hunk 1]
```diff
@@ -10,6 +10,15 @@ inline bool isHiddenTopicOrService(const std::string& name) {
   }
   return name.front() == '_' || name.find("/_") != std::string::npos;
 }
+
+std::string getRosDistro() {
+  const char* rosDistro = std::getenv("ROS_DISTRO");
+  if (rosDistro == nullptr) {
+    return "unknown";
```
Comment (achim-k) [inline ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp:17]: I think it would be good to log a warning if that happens

**Review by achim-k — COMMENTED** [review]
LGTM. Could you apply the changes also for the ROS1 node? Thanks!

File: ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp (hunk 1 above)
Comment (mvukov) [reply ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp:17]: OK, will try to make that happen. Do the bridge and/or studio rely on this env var?

Comment (achim-k) [reply ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp:17]: Yes, Studio uses it to [determine the profile and to load known datatypes](https://github.com/foxglove/studio/blob/79b42ca3e7ad20cc225b094dd791d9e273edf14f/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L285-L304). This is e.g. necessary for publishing messages from Studio

Comment (defunctzombie) [reply ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp:17]: Also worth considering is a fatal error here. This variable should be set for good operation.

**Comment by defunctzombie** [comment]
Closing this due to lack of activity from the last feedback. We are happy to re-visit if you pick this up at a later time.

**Pull request closed without merging**